### PR TITLE
Tweaks to the documentation, build system and bending test

### PR DESCRIPTION
**Pull request by nmnobre**
Hi @bsrunnels,

Me and @karthichockalingam, who I believe you have exchanged some emails with, have tried to compile Alamo and to run a simple cantilever problem.
While doing the former, we found some typos in the documentation and some quirks in the build system we had to work around since we aren't using MPICH. This PR includes a fix which should apply whether you are using MPICH or not.
Unfortunately, even though all tests passed, we still weren't able to run the cantilever problem. We've tried to fix the inputs file to the point where the problem seems to run and the solver seems to iterate but plotting the solution shows no deformation. Is this problem still a work in progress or should we be able to run it with a few extra tweaks?

Cheers,
-Nuno

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -303,11 +303,11 @@ if args.comp == 'icc':
     else:          f.write("CXX_COMPILE_FLAGS += -Ofast -ipo\n") # -inline-forceinline
 
 #
-# MPICH
+# MPI
```
Comment (bsrunnels) [inline configure:306]: Please leave specific to MPICH. Only mpich or mvapich are supported (i.e. not openmpi). This is true of amrex as well. The documentation should reflect the mpich dependency.

File: configure [hunk 2]
```diff
@@ -285,10 +285,10 @@ elif (args.comp == "clang"): f.write("COMP = CLANG\n")
 elif (args.comp == "intel"): f.write("COMP = INTEL\n")
 message("Comp-Type", str(args.comp))
 shellmessage("Comp-Version",args.comp+" --version")
-shellmessage("Comp-MPI","mpicxx -cxx="+args.comp+" -show")
-f.write("CC = mpicxx -cxx="+args.comp+"\n")
+shellmessage("Comp-MPI","mpicxx -show")
```
Comment (bsrunnels) [inline configure:288]: What is the reason for this change? Users may desire to use another compiler (intel or clang) and this would remove that information from the output.

**Review by bsrunnels — CHANGES_REQUESTED** [review]

**Comment by bsrunnels** [comment]
@nmnobre Many of the tests in `./tests/` do not work and need to be removed as you noted. I am not sure whether the cantilever problem works (or has ever worked) but can take a look.

File: configure (hunk 1 above)
Comment (nmnobre) [reply configure:306]: Hi @bsrunnels!

This isn't true of amrex? Why wouldn't we expand to more/all MPI libraries?

Comment (bsrunnels) [reply configure:306]: @nmnobre I agree that it is ideal to support all, but parallelization does not always work correctly when compiled with openmpi. A while back (several years ago) we concluded that openmpi was to blame and dropped alamo support for it. 
I don't think that there's anything preventing you using openmpi (i.e. if you have that installed instead of mpich then it should use it automatically), but keeping mpich specified in the docs is helpful to encourage use of mpich and avoid future errors.

File: configure (hunk 2 above)
Comment (nmnobre) [reply configure:288]: `-cxx` is specific to the mpich wrapper. I forgot to mention we could use `MPICH_CXX`, `OMPI_CXX` and `I_MPI_CXX` instead. I can add this if I manage to convince you to generalise the compilation flow.

**Comment by nmnobre** [comment]
> @nmnobre Many of the tests in `./tests/` do not work and need to be removed as you noted. I am not sure whether the cantilever problem works (or has ever worked) but can take a look.

I never meant to say they should be removed. I'd rather get them fixed. 🙂

**Comment by bsrunnels** [comment]
> I never meant to say they should be removed. I'd rather get them fixed. slightly_smiling_face

That would be great. Adding additional tests is a huge help. I assume you've already looked at the other inputs to see how to write headers that trigger the regression test system. 

One thing to keep in mind, with the cantilever test specifically, is that the solver sometimes struggles with point loads applied at corners. For instance if you wish to apply a point load at the end of a beam, this will often produce divergent behavior. Instead it is recommended to apply a load on the face or to use a `BC` to apply a regularized (smoothed) load at the end.

File: configure [hunk 3]
```diff
@@ -280,36 +280,33 @@ elif (args.eigen != ""):
 #
 if not args.comp in ['g++','icc','clang++']:
     raise Exception(color.red + "ERROR" + color.reset + ": Compiler must be g++, clang++, or icc")
-if (args.comp == "gcc"): f.write("COMP = GCC\n")
-elif (args.comp == "clang"): f.write("COMP = CLANG\n")
-elif (args.comp == "intel"): f.write("COMP = INTEL\n")
```
Comment (bsrunnels) [inline configure:285]: This section should not be removed. The `COMP = ` lines are used to communicate compiler information to the metadata file.

File: configure (hunk 2 above)
Comment (bsrunnels) [reply configure:288]: I do not think this is the right way to do it. Where are the `MPICH_CXX` etc. flags used? I only see them here.
I recommend you do something like: add a `--alternate-mpi` with default value of `none`. Then replace

    f.write("CC = mpicxx -cxx="+args.comp+"\n")

with 

    if not args.alternate_mpi: f.write("CC = mpicxx -cxx="+args.comp+"\n")
    else if args.alternate_mpi = "openmpi":  [ print a CC macro that works for openmpi]
    ....

and additional statements appropriate to each alternate MPI variant.

**Review by bsrunnels — CHANGES_REQUESTED** [review]

File: configure (hunk 3 above)
Comment (nmnobre) [reply configure:285]: None of these conditionals are ever true, check the previous line.

File: configure (hunk 2 above)
Comment (nmnobre) [reply configure]: I wasn't finished, hold on please. :)

Comment (nmnobre) [reply configure]: Looking a bit more like it! I just want to run some tests now ;)

**Comment by nmnobre** [comment]
Hi @bsrunnels,

I think this is now ready for another look.

As discussed, the `configure` tweaks now allow compilation with MPI libraries other than MPICH.
The GitHub workflows still only test with MPICH.

It'd be nice if you could have look at the cantilever beam input file, see if you catch any glaring mistakes.

Cheers :)
